share/p2p/discovery/discovery.go

```diff
@@ -181,16 +175,13 @@ func (d *Discovery) Advertise(ctx context.Context) {
 
 			// we don't want retry indefinitely in busy loop
 			// internal discovery mechanism may need some time before attempts
-			errTimer := time.NewTimer(retryTimeout)
 			select {
-			case <-errTimer.C:
-				errTimer.Stop()
+			case <-time.After(d.params.AdvertiseRetryTimeout):
```

Comment:
why do we do this instead of just setting a short-lived context with params.AdvertiseRetryTimeout for the call to `Advertise` ?

Reply:
The code waits until next retry after `Advertise` returned. Short context does not affect it